### PR TITLE
Implement HTMLStyleElement.disabled attribute

### PR DESCRIPTION
**Pull request by Loirooriol**
https://html.spec.whatwg.org/multipage/#dom-style-disabled

---

- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors
- [X] There are tests for these changes

### CODE REVIEW

File: components/script/dom/htmlstyleelement.rs [hunk 1]
```diff
@@ -280,4 +280,17 @@ impl HTMLStyleElementMethods for HTMLStyleElement {
     fn GetSheet(&self) -> Option<DomRoot<DOMStyleSheet>> {
         self.get_cssom_stylesheet().map(DomRoot::upcast)
     }
+
+    // https://html.spec.whatwg.org/multipage/#dom-style-disabled
```
Comment (mrobinson) [inline components/script/dom/htmlstyleelement.rs:284]:
```suggestion
    /// <https://html.spec.whatwg.org/multipage/#dom-style-disabled>
```

File: components/script/dom/htmlstyleelement.rs [hunk 2]
```diff
@@ -280,4 +280,17 @@ impl HTMLStyleElementMethods for HTMLStyleElement {
...
+    fn Disabled(&self) -> bool {
+        self.get_cssom_stylesheet()
+            .map_or(false, |sheet| sheet.disabled())
+    }
+
+    // https://html.spec.whatwg.org/multipage/#dom-style-disabled
```
Comment (mrobinson) [inline components/script/dom/htmlstyleelement.rs:290]:
```suggestion
    /// <https://html.spec.whatwg.org/multipage/#dom-style-disabled>
```

**Review by mrobinson — APPROVED** [review]
Looks good. Just a couple nits.

File: components/script/dom/htmlstyleelement.rs (hunk 1 above)
Comment (Loirooriol) [reply components/script/dom/htmlstyleelement.rs:284]: OK, I just copied this from the GetSheet above, so I will change that too.

Comment (mrobinson) [reply components/script/dom/htmlstyleelement.rs]: Yeah, this is quite common in the script crate. A lot of the old documentation isn't proper rustdoc.

**Pull request merged**